netutils/config/compliance.py

```diff
@@ -97,17 +99,23 @@ def _is_feature_ordered_compliant(feature_intended_cfg, feature_actual_cfg):
     return False
 
 
-def _open_file_config(cfg_path):
+def _open_file_config(cfg_path: str) -> t.Union[str, bool]:
     """Open config file from local disk."""
     try:
         with open(cfg_path, encoding="utf-8") as filehandler:
             device_cfg = filehandler.read()
     except IOError:
-        return False
+        return False  # This should probably be changed to a exception raise. Causing mypy issues on L183, L184
     return device_cfg.strip()
 
 
-def compliance(features, backup, intended, network_os, cfg_type="file"):
+def compliance(
+    features: t.List[t.Dict[str, t.Union[str, bool, t.List[str]]]],
+    backup: str,
+    intended: str,
+    network_os: str,
+    cfg_type: t.Optional[str] = "file",
```

Comment:
If we supply a default argument for this but make the type `Optional`, do we expect users to explicitly pass `None` to `cfg_type` or can we just omit the `Optional`?